### PR TITLE
Relax query expression keywords

### LINKED ISSUE #32061: Relax keywords used in query expression
**Description:**
Consider keywords used in the query expression as keywords only within the query expression context, such that those names can be used as identifiers in other places. The target keyword set is mentioned below.
- `where`
- `join`
- `order`
- `by`
- `equals`
- `ascending`
- `descending`
- `limit`
- `outer`
- `select`

Spec issue https://github.com/ballerina-platform/ballerina-spec/issues/34

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Fixes #32061

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -4947,6 +4953,28 @@ private STNode parseTerminalExpression(STNode annots, List<STNode> qualifiers, b
         }
     }
 
+    private boolean isValidQueryCtxKeyword(STToken nextToken) {
+        if (!hasAncestorContext(ParserRuleContext.QUERY_EXPRESSION)) {
+            return false;
+        }
```
Comment (lochana-chathura) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:4959]: We can do this at lexer level. When we are inside parsing query expression, we capture those tokens as keyword where as in all other places we capture them as identifiers. 
e.g.
```java
    private STNode parseQueryExpression() {
        this.tokenReader.startMode(ParserMode.QUERY);
        // Query expression parsing
        this.tokenReader.endMode();
    }
```
I think that would make many things simple.

Comment (rdulmina) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java]: Updated

**Comment by lochana-chathura** [comment]
Looks good. Let's fix the build failure.

**Review by lochana-chathura — APPROVED** [review]

**Comment by lochana-chathura** [comment]
@MaryamZi this has to go in the release notes right? @rdulmina Please note.

**Pull request merged**